kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -2614,6 +2614,17 @@ object KyuubiConf {
       .toSequence()
       .createWithDefault(Nil)
 
+  val SERVER_LIMIT_CONNECTIONS_USER_LIMITED_LIST: ConfigEntry[Seq[String]] =
+    buildConf("kyuubi.server.limit.connections.user.limited.list")
+      .doc("The user in the limited list will be limited to connect to kyuubi server, " +
+        "if the user has configured both user.unlimited.list and user.limited.list, " +
+        "the priority of the latter is higher.")
+      .version("1.8.0")
+      .serverOnly
+      .stringConf
+      .toSequence()
```

Comment:
```suggestion
      .toSet()
```
With #5185 introduced set collection support for configs, you could rebase to the master branch and use it.

Reply:
Thank you for your reminder.

Reply:
LGTM after this comment addressed

Reply:
@turboFei hi, do you mean that aside from the above case, my current code implementation is okay? but @pan3793  thinks it's incorrect.

Reply:
I had an offline discussion with @turboFei, and we reached a consensus that:

1. The current implementation is **deny** semantic, which is a good feature but **limited** is not a properly name
2. You could either change the configuration name to `...deny.user.list`, or keep the `...limited.user.list` and change the implementation as I commented above https://github.com/apache/kyuubi/pull/5182#discussion_r1304545505, I think the latter approach is simple, and we can implement the **deny** in another PR.

Reply:
@pan3793 hi, I have made some changes to the code, and I hope I understood what you wanted. Could you please review the code and give me some suggestions? Thank you!

Reply:
@zhuyaogai will take a look tomorrow, have a good weekend :)

Reply:
@pan3793 BTW, I found that is there a problem with the current master branch code? When some users open some sessions and before closing the sessions, this user is added to the `unlimited.list`, then the counter related to this user will not be cleared.

Reply:
I think we don't need to check any limit in `decrement` method, and let the already opened sessions decrement limit counter. Please correct me if I am wrong.

Reply:
> I had an offline discussion with @turboFei, and we reached a consensus that:
> 
> 1. The current implementation is **deny** semantic, which is a good feature but **limited** is not a properly name
> 2. You could either change the configuration name to `...deny.user.list`, or keep the `...limited.user.list` and change the implementation as I commented above [[KYUUBI #4540] [Improvement] Support to limit some user to make connection. #5182 (comment)](https://github.com/apache/kyuubi/pull/5182#discussion_r1304545505), I think the latter approach is simple, and we can implement the **deny** in another PR.

I have similar concerns in these variations. I was confused by the **deny** and **limited**.
As for me, a limited user list is not the contrast to an unlimited user, and only a feature for  deny users list is necessary.
What's more I would like to see when the server reject user connection , this feature should be part of authentication ,such as User filerter,  but not an injected mechanism in such Limiter. Or did I miss any information for better understanding?

I would suggest, 
1. Add detailed and summarised design and key points to this PR description, as a simple link to the original issue is not enough
2. Add constructive discussion in proposed change and the reason for possible approaches

I'm fine if we accept this feature, especially when some of us have a mutual understanding of enough information.